### ISSUE TITLE
'az rest' with param '--resource' ending without '/'

### ISSUE THREAD

**Issue by bpelikan** [Azure/azure-cli#10199]
## Describe the bug

**Command Name**
`az rest`

**Errors:**
```
400 Client Error: Bad Request for url: http://localhost:50342/oauth2/token
Traceback (most recent call last):
python3.6/site-packages/knack/cli.py, ln 206, in invoke
    cmd_result = self.invocation.execute(args)
cli/core/commands/__init__.py, ln 603, in execute
    raise ex
cli/core/commands/__init__.py, ln 661, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
cli/core/commands/__init__.py, ln 654, in _run_job
    six.reraise(*sys.exc_info())
...
python3.6/site-packages/msrestazure/azure_active_directory.py, ln 486, in get_msi_token
    result.raise_for_status()
python3.6/site-packages/requests/models.py, ln 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://localhost:50342/oauth2/token
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- `az rest --uri {} --method {} --headers {} --resource "https://management.azure.com"`

## Expected Behavior
Could it work like with param ending with `/` or although to return some more information that param `--resource` is incorrect?

## Environment Summary
```
Linux-4.15.0-1052-azure-x86_64-with-debian-stretch-sid
Python 3.6.5
Shell: bash

azure-cli 2.0.70

```
## Additional Context

Working fine with `--resource` ending `/`:
- `az rest --uri {} --method {} --headers {} --resource "https://management.azure.com/"`

**Comment by jiasli** [comment]
I am able to repro on Cloud Shell. This seems to be related to #9345. On a local CLI, both `"https://management.azure.com/"` and `"https://management.azure.com"` (with or without `/`) works fine. Could you try that on a local CLI?

**Comment by bpelikan** [comment]
Yes, on a local CLI this is working fine.

**Comment by rich617** [comment]
Having worked on acquiring resource tokens in the CLI this last month, I can tell you this:

1. The reason it doesn't work without the trailing / is that ARM code doesn't have a mapping registered for the resource without the trailing / for its GetDelegatedToken API. 
2. If you really want it to work without the trailing /, you can work around this issue by doing "az login" in Cloud Shell, and following the instructions to paste the auth code into a browser window. After doing this, Cloud Shell will issue tokens identically to on-prem CLI. (Note: This behavior cannot be built-in to Cloud Shell as an automatic thing.)
3. You are taking some chances by using the resource without the trailing /. Although the audience in the resulting token will work for token validation, and so many things will work, any on-behalf-of flows with the / missing will FAIL because the audience of the token needs to match the subject name in the 1st party app certificate exactly, including the trailing /. 

In short, the simple solution is to just put the trailing / if Cloud Shell requires it. 

Also, FYI, if you do "az cloud show", I believe the endpoint will be listed with the trailing /. 

I do agree that the error message could be more informative than a stack trace. That would be a possible code change for this issue. 

Also, I recently added code to "az account get-access-token" to allow the resource to be specified as a more user-friendly --resource-type. So, instead of --resource https://management.azure.com/, you can say --resource-type arm. I wasn't aware of the "az rest" command, and it would be good to add the same feature there.

**Comment by jiasli** [comment]
We have fixed Cloud Shell to accept both `https://management.azure.com/` and 
`https://management.azure.com`. Please try again.
